app/src/main/java/com/medmapper/v33001/dto/Medicine.kt

```diff
@@ -3,13 +3,16 @@ package com.medmapper.v33001.dto
 import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.PrimaryKey
+import androidx.room.TypeConverters
+import com.medmapper.v33001.dao.DateConverter
 
 @Entity(tableName = "Medicine")
+@TypeConverters(DateConverter::class)
 data class Medicine(
     @PrimaryKey(autoGenerate = true) val medID: Int,
     @ColumnInfo(name = "name") val name: String?,
     @ColumnInfo(name = "strength") val strength: String?,
-    @ColumnInfo(name = "start date") val startDate: String?,
+    @ColumnInfo(name = "start_date") val startDate: String?,
```

Comment:
Column names should be lowercase, with words separated by underscores ("_").

Source: https://developer.android.com/training/data-storage/room/defining-data